service/history/ndc/hsm_state_replicator.go

```diff
@@ -183,10 +183,19 @@ func (r *HSMStateReplicatorImpl) syncHSMNode(
 		incomingNodePath := incomingNode.Path()
 		currentNode, err := currentHSM.Child(incomingNodePath)
 		if err != nil {
-			// 1. Already done history resend if needed before,
-			// and node creation today always associated with an event
-			// 2. Node deletion is not supported right now.
-			// Based on 1 and 2, node should always be found here.
+			// The node may not be found if:
+			// 1. The state machine was deleted (e.g. terminal state cleanup)
+			// 2. We're missing events that created this node
```

Comment:
That's not true based on the comment that you deleted.

> Already done history resend if needed before,
			// and node creation today always associated with an event

I would also clarify that creation and deletion are always associated with an event.